routes.go

```diff
@@ -55,19 +57,57 @@ type MetricMiddlewareOut struct {
 	Health  alice.Chain `name:"middleware_health_metrics"`
 }
 
-func buildPrimaryRoutes(in PrimaryRouterIn) {
+func handlePrimaryEndpoint(in PrimaryRouterIn) {
 	in.Router.Use(in.AuthChain.Then)
 	in.Router.PathPrefix("/").Handler(httpaux.ConstantHandler{StatusCode: http.StatusOK})
 }
 
-func buildMetricsRoutes(in MetricsRoutesIn) {
-	if in.Handler != nil {
-		in.Router.Handle("/metrics", in.Handler).Methods("GET")
-	}
+func handledMetricEndpoint(in MetricsRoutesIn) {
+	in.Router.Handle("/metrics", in.Handler).Methods("GET")
 }
 
 func metricMiddleware(bundle touchhttp.ServerBundle) (out MetricMiddlewareOut) {
 	out.Primary = alice.New(bundle.ForServer("server_primary").Then)
 	out.Health = alice.New(bundle.ForServer("server_health").Then)
 	return
 }
+
+func provideServers() fx.Option {
+	return fx.Options(
+		fx.Provide(
+			metricMiddleware,
+		),
+		arrangehttp.Server{
+			Name: "server_primary",
+			Key:  "servers.primary",
+			Inject: arrange.Inject{
+				PrimaryMMIn{},
+			},
+		}.Provide(),
+		arrangehttp.Server{
+			Name: "server_health",
+			Key:  "servers.health",
+			Inject: arrange.Inject{
+				HealthMMIn{},
+			},
+			Invoke: arrange.Invoke{
+				func(r *mux.Router) {
+					r.Handle("/health", httpaux.ConstantHandler{
+						StatusCode: http.StatusOK,
+					}).Methods("GET")
+				},
+			},
+		}.Provide(),
+		arrangehttp.Server{
+			Name: "server_metrics",
+			Key:  "servers.metrics",
+		}.Provide(),
+		fx.Invoke(
+			serverValidator{Key: "servers.primary"}.Validate,
+			serverValidator{Key: "servers.metrics"}.Validate,
+			serverValidator{Key: "servers.health"}.Validate,
```

Comment:
What does this do? I know it was already there in main.go, just curious. 🙂

Reply:
It checks that the host used in the `address` field of each of these servers is always a loopback address. The reason being that `authbaton` should only be reachable from a reverse proxy running on the same host